MatrixSDK/JSONModels/Login/MXLoginSSOIdentityProvider.m

```diff
@@ -0,0 +1,49 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "MXLoginSSOIdentityProvider.h"
+
+@interface MXLoginSSOIdentityProvider()
+
+@property (nonatomic, readwrite) NSString *identifier;
+@property (nonatomic, readwrite) NSString *name;
+@property (nonatomic, readwrite, nullable) NSString *icon;
+
+@end
+
+@implementation MXLoginSSOIdentityProvider
+
++ (id)modelFromJSON:(NSDictionary *)JSONDictionary
+{
+    NSString *identifier;
+    NSString *name;
+    
+    MXJSONModelSetString(identifier, JSONDictionary[@"id"]);
+    MXJSONModelSetString(name, JSONDictionary[@"name"]);
+    
+    MXLoginSSOIdentityProvider *identityProvider = [MXLoginSSOIdentityProvider new];
+    
+    if (identityProvider && identifier && name)
```

Comment:
Have two concerns.
1. We're declaring that `identifier` and `name` will be non-nil with `NS_ASSUME_NONNULL_BEGIN`, but if they are nil after parsing, we still return them nil here.
2. This introduces some logic in a model class, which doesn't seem to be a good idea.

Reply:
Yes I should return nil in this case. In the MSC identifier && name have to be non optional.

Reply:
Cool, thanks.